src/plugins/vis_type_xy/public/utils/render_all_series.tsx

```diff
@@ -71,9 +72,17 @@ export const renderAllSeries = (
       interpolate,
       type,
     }) => {
-      const yAspects = aspects.y.filter(
-        ({ aggId, accessor }) => aggId?.includes(paramId) && accessor !== null
-      );
+      const yAspects = aspects.y.filter(({ aggId, aggType, accessor }) => {
+        if (
+          aggType === METRIC_TYPES.PERCENTILES ||
+          aggType === METRIC_TYPES.PERCENTILE_RANKS ||
+          aggType === METRIC_TYPES.STD_DEV
+        ) {
+          return aggId?.includes(paramId) && accessor !== null;
+        } else {
+          return aggId === paramId && accessor !== null;
+        }
+      });
```

Comment:
Is this always the case? Would it make sense to just check if `Array.isArray(aggId)` then check for the `paramId` accordingly?

Reply:
aggId is not an array in these three cases, it is a string that contains the paramId. For example for STD_DEV is 1_std.low.